templates/mail/team_invite.tmpl

```diff
@@ -4,10 +4,15 @@
 	<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
 	<meta name="format-detection" content="telephone=no,date=no,address=no,email=no,url=no">
 </head>
-{{$invite_url := printf "%sorg/invite/%s" AppUrl (QueryEscape .Invite.Token)}}
 <body>
 	<p>{{.locale.Tr "mail.team_invite.text_1" (DotEscape .Inviter.DisplayName) (DotEscape .Team.Name) (DotEscape .Organization.DisplayName) | Str2html}}</p>
-	<p>{{.locale.Tr "mail.team_invite.text_2"}}</p><p><a href="{{$invite_url}}">{{$invite_url}}</a></p>
+	{{if .UserAccountExists}}
+		{{$login_url := printf "%suser/login?redirect_to=%s" AppUrl (QueryEscape (printf "/org/invite/%s" .Invite.Token))}}
+		<p>{{.locale.Tr "mail.team_invite.text_2"}}</p><p><a href="{{$login_url}}">{{$login_url}}</a></p>
+	{{else}}
+		{{$signup_url := printf "%suser/sign_up?redirect_to=%s" AppUrl (QueryEscape (printf "/org/invite/%s" .Invite.Token))}}
+		<p>{{.locale.Tr "mail.team_invite.text_2"}}</p><p><a href="{{$signup_url}}">{{$signup_url}}</a></p>
+	{{end}}
```

Comment:
Can this computation be shifted to controller level? just use the variable in the from context.